### PR TITLE
[2024-08-24] Bump dependency updates identified by dependabot

### PR DESCRIPTION
**Pull request by henrybear327**
This pull request completes this week's etcd dependency updates following our [dependency roster](https://docs.google.com/spreadsheets/d/1jodHIO7Dk2VWTs1IRnfMFaRktS9IH8XRyifOnPdSY8I/edit#gid=1394774387) and [dependency management instructions](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md).

Summary of actions

Bumped
- bump github.com/grpc-ecosystem/grpc-gateway/v2 from 2.21.0 to 2.22.0
  - https://github.com/etcd-io/etcd/pull/18470
  - https://github.com/etcd-io/etcd/pull/18467
- bump github.com/prometheus/client_golang from 1.19.1 to 1.20.0
  - https://github.com/etcd-io/etcd/pull/18469
  - https://github.com/etcd-io/etcd/pull/18466
- bump honnef.co/go/tools from 0.4.7 to 0.5.1
  - https://github.com/etcd-io/etcd/pull/18463

Fully indirect
- bump github.com/polyfloyd/go-errorlint from 1.5.2 to 1.6.0
  - https://github.com/etcd-io/etcd/pull/18465
- bump github.com/tomarrell/wrapcheck/v2 from 2.8.3 to 2.9.0
  - https://github.com/etcd-io/etcd/pull/18464

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tools/mod/go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module go.etcd.io/etcd/tools/v3
 
-go 1.22
+go 1.22.1
 
-toolchain go1.22.6
+toolchain go1.23.0
```
Comment (ivanvc) [inline tools/mod/go.mod:5]: Hi @henrybear327, I think the verify job is failing due to this change.

Comment (henrybear327) [reply tools/mod/go.mod]: Confirmed to be the issue! The new commit is pushed :) Thanks @ivanvc

**Review by jmhbnz — APPROVED** [review]
LGTM - Much appreciated @henrybear327 🙏🏻

**Review by ivanvc — APPROVED** [review]
LGTM. Thanks, Henry.

**Pull request merged**